magento/pwa-studio
# Order confirmation page multi refresh fix

## Description

`useCheckoutPage` talon has a `useEffect` that fetches checkout details everytime the `cartId` changes. When the user places an order, a new `cartId` will be created because the old one will be purged. Due to this, the `useEffect` was being triggered and in the process `isLoading` was changing to `true`. Because of this, the user will see a loading screen for a bit and the order confirmation page will be reloaded. This is un-intended behavior.

To fix this, I have added a check to only call the checkout details query if the order has not been placed.

The talon involved in this PR does not have a tests file defined. I need to create and add some tests.

## Related Issue
Closes [PWA-580](https://jira.corp.magento.com/browse/PWA-580)

### Verification Stakeholders
@sirugh 

### Verification Steps
1. Go to the deployed app and try to check out a product.
2. When the order confirmation page loads, there should not be page flashes.

## Checklist
